### PR TITLE
get csv for agency data from protected endpoint

### PR DESCRIPTION
**Pull request by nichelle-hall**
## Description of the change

Updates where we fetch the agencies uploaded data to confirm that only logged-in users can access CSVs containing all uploaded data. 

## Related issues

[Companion BE PR](https://github.com/Recidiviz/recidiviz-data/pull/24903)

### Development

**This box MUST be checked by the submitter prior to merging**:
- [ ] **Double- and triple-checked that there is no Personally Identifiable Information (PII) being mistakenly added in this pull request**

These boxes should be checked by the submitter prior to merging:
- [ ] Tests have been written to cover the code changed/added as part of this pull request

### Code review

These boxes should be checked by reviewers prior to merging:

- [ ] This pull request has a descriptive title and information useful to a reviewer
- [ ] This pull request has been moved out of a Draft state, has no "Work In Progress" label, and has assigned reviewers
- [ ] Potential security implications or infrastructural changes have been considered, if relevant

### CODE REVIEW

File: common/utils/downloadHelpers.ts [hunk 1]
```diff
@@ -23,7 +23,7 @@ export const downloadFeedData = async (
   filename: string
 ) => {
   const a = document.createElement("a");
-  a.href = `/feed/${agencyId}?system=${system}&metric=${filename}`;
+  a.href = `/api/feed/${agencyId}?system=${system}&metric=${filename}`;
```
Comment (lilidworkin) [inline common/utils/downloadHelpers.ts:26]: This method is in the `common` repo, which suggests to me there's a chance this is used by both Agency Dashboard and Publisher. If that's the case, we actually want Agency Dashboard to continue using the unauthenticated version, with only published data.

File: common/utils/downloadHelpers.ts [hunk 2]
```diff
@@ -20,22 +20,28 @@ import { Metric } from "@justice-counts/common/types";
 export const downloadFeedData = async (
   system: string,
   agencyId: number | string,
-  filename: string
+  filename: string,
+  isPublic: boolean
```
Comment (mxosman) [inline common/utils/downloadHelpers.ts:24]: QQ - what does the `isPublic` refer to?

**Review by mxosman — APPROVED** [review]
LGTM, Nichelle! Just had one question.

**Review by lilidworkin — APPROVED** [review]
Awesome, thanks for making my suggested change @nichelle-hall !

File: common/utils/downloadHelpers.ts (hunk 2 above)
Comment (nichelle-hall) [reply common/utils/downloadHelpers.ts:24]: isPublic refers to if the information will be share publicly (i.e agency dashboard versus publisher)

Comment (mxosman) [reply common/utils/downloadHelpers.ts:24]: Ah - gotcha! Not to throw a nit post-approval, but might be worth adding a comment for clarity.

Comment (nichelle-hall) [reply common/utils/downloadHelpers.ts:24]: No problem! I think taking time to write a comment is always worth it.

File: common/utils/downloadHelpers.ts [hunk 3]
```diff
@@ -20,22 +20,34 @@ import { Metric } from "@justice-counts/common/types";
...
   const a = document.createElement("a");
-  a.href = `/feed/${agencyId}?system=${system}&metric=${filename}`;
+  let url = `/feed/${agencyId}?system=${system}&metric=${filename}`;
+  if (isPublic === false) {
+    // the isPublic determines where we fetch the an agency's data.
+    // If the information WILL NOT be shared publicly (i.e isPublic === false),
+    // data is requested from a protected endpoint that will return
+    // all uploaded data for an agency. If the data WILL be shared
+    // publicly, data is fetched from a public endpoint that will
+    // only return published data.
```
Comment (mxosman) [inline common/utils/downloadHelpers.ts:34]: LOVE this comment - thank you so much, Nichelle!!

**Pull request merged**